owncloudApp/src/main/java/com/owncloud/android/presentation/common/DrawerViewModel.kt

```diff
@@ -25,65 +25,43 @@ package com.owncloud.android.presentation.common
 
 import android.accounts.Account
 import android.content.Context
-import androidx.lifecycle.LiveData
-import androidx.lifecycle.MediatorLiveData
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import com.owncloud.android.R
 import com.owncloud.android.data.providers.LocalStorageProvider
 import com.owncloud.android.domain.user.model.UserQuota
-import com.owncloud.android.domain.user.usecases.GetStoredQuotaUseCase
+import com.owncloud.android.domain.user.usecases.GetStoredQuotaAsStreamUseCase
 import com.owncloud.android.domain.user.usecases.GetUserQuotasUseCase
-import com.owncloud.android.domain.utils.Event
-import com.owncloud.android.extensions.ViewModelExt.runUseCaseWithResult
 import com.owncloud.android.presentation.authentication.AccountUtils
 import com.owncloud.android.providers.ContextProvider
 import com.owncloud.android.providers.CoroutinesDispatcherProvider
 import com.owncloud.android.usecases.accounts.RemoveAccountUseCase
+import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.launch
 import timber.log.Timber
 
 class DrawerViewModel(
-    private val getStoredQuotaUseCase: GetStoredQuotaUseCase,
+    getStoredQuotaAsStreamUseCase: GetStoredQuotaAsStreamUseCase,
     private val removeAccountUseCase: RemoveAccountUseCase,
     private val getUserQuotasUseCase: GetUserQuotasUseCase,
     private val localStorageProvider: LocalStorageProvider,
     private val coroutinesDispatcherProvider: CoroutinesDispatcherProvider,
     private val contextProvider: ContextProvider,
+    accountName: String,
 ) : ViewModel() {
 
-    private val _userQuota = MediatorLiveData<Event<UIResult<UserQuota?>>>()
-    val userQuota: LiveData<Event<UIResult<UserQuota?>>> = _userQuota
-
-    fun getStoredQuota(
-        accountName: String
-    ) = runUseCaseWithResult(
-        coroutineDispatcher = coroutinesDispatcherProvider.io,
-        requiresConnection = false,
-        showLoading = true,
-        liveData = _userQuota,
-        useCase = getStoredQuotaUseCase,
-        useCaseParams = GetStoredQuotaUseCase.Params(accountName = accountName)
-    )
+    val userQuota: Flow<UserQuota> = getStoredQuotaAsStreamUseCase(GetStoredQuotaAsStreamUseCase.Params(accountName))
```

Comment:
Hm, not sure if I like this... I think it would be better being encapsulated with a private variable. Otherwise everytime we access `userQuota`, the use case might be executed, and that's not what we want. Check for example `SettingsPictureUploadsViewModel` to see how this is done with the `pictureUploads` variable 👍

Reply:
Done! 😄